src/interfaces.ts

```diff
@@ -28,36 +28,81 @@ export interface SpanielObserverInit {
   USE_NATIVE_IO?: boolean;
 }
 
+export interface TimeCompat {
+  highResTime: number;
+  unixTime: number;
+}
+
 export interface SpanielRecord {
   target: SpanielTrackedElement;
   payload: any;
   thresholdStates: SpanielThresholdState[];
-  lastSeenEntry: IntersectionObserverEntry | null;
+  lastSeenEntry: MaybeInternalIntersectionObserverEntry | null;
 }
 
 export interface SpanielThresholdState {
   lastSatisfied: Boolean;
-  lastEntry: IntersectionObserverEntry | null;
+  lastEntry: MaybeInternalIntersectionObserverEntry | null;
   threshold: SpanielThreshold;
-  lastVisible: number;
+  lastVisible: TimeCompat;
   visible: boolean;
   timeoutId?: number;
 }
 
 export interface SpanielIntersectionObserverEntryInit {
-  time: DOMHighResTimeStamp;
-  rootBounds: ClientRect;
-  boundingClientRect: ClientRect;
-  intersectionRect: ClientRect;
+  highResTime: DOMHighResTimeStamp;
+  unixTime: number;
+  rootBounds: SpanielRect;
+  boundingClientRect: SpanielRect;
+  intersectionRect: SpanielRect & ClientRect;
   target: SpanielTrackedElement;
 }
 
-export interface SpanielObserverEntry extends IntersectionObserverEntryInit {
+export interface SpanielRect extends Partial<DOMRectReadOnly> {
```

Comment:
Isn't this just `Readonly<Required<DOMRectInit>>`?
And why do we make this change for the duration fix?

Reply:
The previous code was not accurately typing things. SpanielObserver operates of entries from the intersection observer, which can either be the native intersection observer or the spaniel intersection observer. But the entry was always being typed as the native `IntersectionObserverEntry`, whereas the entries from the spaniel intersection observer polyfill are not exactly the same because they are missing a few sub fields. When I addressed this issue, there was a cascading effect. Basically the typings have to respect the fact that the entries could be from native or polyfill IO.